### ISSUE TITLE
HaxePsiBuilder is trying to work with gradle files?

### ISSUE THREAD

**Issue by mayakwd** [HaxeFoundation/intellij-haxe#695]
```
Invalid PSI ASTWrapperPsiElement(FILE) of class com.intellij.extapi.psi.ASTWrapperPsiElement for AST Element(FILE) of class com.intellij.psi.impl.source.tree.FileElement
java.lang.AssertionError: Invalid PSI ASTWrapperPsiElement(FILE) of class com.intellij.extapi.psi.ASTWrapperPsiElement for AST Element(FILE) of class com.intellij.psi.impl.source.tree.FileElement
	at com.intellij.psi.impl.source.tree.SharedImplUtil.getContainingFile(SharedImplUtil.java:72)
	at com.intellij.lang.impl.PsiBuilderImpl.<init>(PsiBuilderImpl.java:129)
	at com.intellij.plugins.haxe.lang.parser.HaxePsiBuilder.<init>(HaxePsiBuilder.java:62)
	at com.intellij.plugins.haxe.lang.parser.HaxePsiBuilderFactoryImpl.createBuilder(HaxePsiBuilderFactoryImpl.java:35)
	at com.intellij.psi.tree.ILazyParseableElementType.doParseContents(ILazyParseableElementType.java:86)
	at com.intellij.psi.tree.ILazyParseableElementType.parseContents(ILazyParseableElementType.java:80)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.ensureParsed(LazyParseableElement.java:192)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:240)
	at com.intellij.psi.impl.source.tree.CompositeElement.countChildren(CompositeElement.java:466)
	at com.intellij.psi.impl.source.tree.CompositeElement.getChildren(CompositeElement.java:411)
	at com.intellij.plugins.haxe.util.ASTNodeWalker.walk(ASTNodeWalker.java:43)
	at com.intellij.plugins.haxe.util.ASTNodeWalker.walk(ASTNodeWalker.java:45)
	at com.intellij.plugins.haxe.util.ASTNodeWalker.walk(ASTNodeWalker.java:45)
	at com.intellij.plugins.haxe.lang.parser.HaxePsiBuilder.getTreeBuilt(HaxePsiBuilder.java:93)
	at org.jetbrains.plugins.groovy.lang.parser.GroovyParser.parse(GroovyParser.java:81)
	at com.intellij.psi.tree.ILazyParseableElementType.doParseContents(ILazyParseableElementType.java:88)
	at com.intellij.psi.tree.IFileElementType.parseContents(IFileElementType.java:49)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.ensureParsed(LazyParseableElement.java:192)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:240)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:39)
	at com.intellij.psi.impl.source.tree.SharedImplUtil.getFirstChild(SharedImplUtil.java:52)
	at com.intellij.psi.impl.source.PsiFileImpl.getFirstChild(PsiFileImpl.java:851)
	at org.jetbrains.plugins.groovy.lang.psi.impl.GroovyFileBaseImpl.acceptChildren(GroovyFileBaseImpl.java:159)
	at org.jetbrains.plugins.groovy.lang.psi.GroovyRecursiveElementVisitor.visitElement(GroovyRecursiveElementVisitor.java:27)
	at org.jetbrains.plugins.groovy.lang.psi.GroovyElementVisitor.visitFile(GroovyElementVisitor.java:65)
	at org.jetbrains.plugins.groovy.lang.psi.impl.GroovyFileBaseImpl.accept(GroovyFileBaseImpl.java:154)
	at org.jetbrains.plugins.grails.lang.gsp.gspIndex.GspIncludeIndex.map(GspIncludeIndex.java:255)
	at org.jetbrains.plugins.grails.lang.gsp.gspIndex.GspIncludeIndex.map(GspIncludeIndex.java:51)
```

**Comment by EricBishton** [comment]
No, but yes.   What I was trying to fix: Find parsing errors that occur within the Haxe plugin.  I was doing that by capturing the errors as they occur when parsing.

A question: do you think that this change caused the ASTWrapper issue or are you simply surprised that it was on the stack?

The thing of it is that the PsiBuilder class is a system-wide service.  A plugin can override it, but it overrides it for *everything*.  And there is no chain of builders like there are for other types of services (e.g. parsers).

Now, I built a simple wrapper class.  The only thing it does different is, IF the debug mode is set (which it should NOT normally be), it walks the tree and reports errors before returning the built Psi tree to the caller.  However, that seems to be causing a bit of trouble because the tree walk (using the normal PSI routines) tries to re-parse the lazy nodes and that creates an issue with unit tests.  But it may be causing issues elsewhere.  That said, I really don't understand why it would cause an issue if debugging is disabled because the service doesn't do the tree walk.

So, if this continues to cause trouble, what I'm going to do is remove the service declaration from the plugin.xml and add a comment about how to enable it for when we need to use it for debugging.  That way, it won't normally affect people (even though it shouldn't now, unless there is another plugin trying to map the PsiBuilder itself).

**Comment by EricBishton** [comment]
I didn't realize that I'd left the "static { LOG.setLevel(Level.DEBUG));" line in HaxePsiBuilder.  That's not good.  I'll fix it right away.

**Comment by EricBishton** [comment]
Fixed with aa05f53d6ef394ca51d794d7d99bed2b781b6716
